### PR TITLE
 Remove 3 standard controller (route autoscaler metrics )

### LINKED ISSUE #1080: [Feature] remove controllers in standard.oam.dev to catalog
**Is your feature request related to a problem? Please describe.**

We have four controllers in standard.oam.dev, that are not enabled by default now, we should move it out from vela-core.

1. autoscaler
2. metrics
3. podspecworkload
4. routes

**Describe the solution you'd like**
1. Move to catalog https://github.com/oam-dev/catalog
2. let them become 4 charts
3. create a chart repository for catalog, let the user can download chart from that。（e.g. https://github.com/wonderflow/crossplane-oam-sample/tree/master/archives）
4. update the definition file related, add helm install section(e.g. https://github.com/oam-dev/catalog/blob/master/registry/kubewatch.yaml#L14-L19)

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #1130: launch kubevela failed for CRD not found
Since `ServiceMonitor.monitoring.coreos.com` is already removed from kubevela chart, when launch kubevela,  controller-manager crashed for metrics controller depends on watching `servicemonitors`.
```
2021-03-03T20:11:20.735+0900	INFO	controller	Starting Controller	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "ApplicationConfiguration", "controller": "oam/applicationconfiguration.core.oam.dev"}
I0303 20:11:21.680725   19627 request.go:621] Throttling request took 1.0421498s, request: GET:https://127.0.0.1:36763/apis/apiregistration.k8s.io/v1beta1?timeout=32s
2021-03-03T20:11:22.982+0900	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "ServiceMonitor.monitoring.coreos.com", "error": "no matches for kind \"ServiceMonitor\" in version \"monitoring.coreos.com/v1\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/home/roywang/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start
	/home/roywang/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/source/source.go:117
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/home/roywang/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:140
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/home/roywang/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:175
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
	/home/roywang/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/manager/internal.go:661
2021-03-03T20:11:22.982+0900	INFO	controller	Starting workers	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "ManualScalerTrait", "controller": "oam/manualscalertrait", "worker count": 1}
2021-03-03T20:11:22.982+0900	INFO	controller	Stopping workers	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "ManualScalerTrait", "controller": "oam/manualscalertrait"}
2021-03-03T20:11:22.982+0900	INFO	controller	Starting Controller	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "ApplicationDeployment", "controller": "applicationdeployment"}
2021-03-03T20:11:22.982+0900	INFO	controller	Starting workers	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "HealthScope", "controller": "oam/healthscope.core.oam.dev", "worker count": 1}
2021-03-03T20:11:22.982+0900	INFO	controller	Stopping workers	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "HealthScope", "controller": "oam/healthscope.core.oam.dev"}2021-03-03T20:11:20.735+0900	INFO	controller	Starting Controller	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "ApplicationConfiguration", "controller": "oam/applicationconfiguration.core.oam.dev"}
I0303 20:11:21.680725   19627 request.go:621] Throttling request took 1.0421498s, request: GET:https://127.0.0.1:36763/apis/apiregistration.k8s.io/v1beta1?timeout=32s
2021-03-03T20:11:22.982+0900	ERROR	controller-runtime.source	if kind is a CRD, it should be installed before calling Start	{"kind": "ServiceMonitor.monitoring.coreos.com", "error": "no matches for kind \"ServiceMonitor\" in version \"monitoring.coreos.com/v1\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/home/roywang/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/source.(*Kind).Start
	/home/roywang/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/source/source.go:117
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
	/home/roywang/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:140
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/home/roywang/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/internal/controller/controller.go:175
sigs.k8s.io/controller-runtime/pkg/manager.(*controllerManager).startRunnable.func1
	/home/roywang/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.2/pkg/manager/internal.go:661
2021-03-03T20:11:22.982+0900	INFO	controller	Starting workers	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "ManualScalerTrait", "controller": "oam/manualscalertrait", "worker count": 1}
2021-03-03T20:11:22.982+0900	INFO	controller	Stopping workers	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "ManualScalerTrait", "controller": "oam/manualscalertrait"}
2021-03-03T20:11:22.982+0900	INFO	controller	Starting Controller	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "ApplicationDeployment", "controller": "applicationdeployment"}
2021-03-03T20:11:22.982+0900	INFO	controller	Starting workers	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "HealthScope", "controller": "oam/healthscope.core.oam.dev", "worker count": 1}
2021-03-03T20:11:22.982+0900	INFO	controller	Stopping workers	{"reconcilerGroup": "core.oam.dev", "reconcilerKind": "HealthScope", "controller": "oam/healthscope.core.oam.dev"}
```

Shall we add the CRD back to the chart?

### PR DESCRIPTION
**Pull request by wangyikewxgm**
fix #1080 and will fix #1130  3 standard controller migrated to catalog https://github.com/oam-dev/catalog
- [x] remove 4 standard controllers code(route autoscaler metrics)
- [x] remove related crd and resources
- [x] add docs for install them with chart

### CODE REVIEW

**Comment by wonderflow** [comment]
please also refactor the doc in this PR

**Comment by wonderflow** [comment]
please also cherry-pick to `release-0.3` after merge

File: docs/en/developers/extensions/set-metrics.md [hunk 1]
```diff
@@ -3,12 +3,37 @@
...
+2. Update the chart repo
+    ```shell script
+    helm repo update
+    ```
+
+3. Install metrics trait controller
+    ```shell script
+    helm install --create-namespace -n vela-system metricstrait oam.catalog/metricstrait
+
+##Setting metrics policy
```
Comment (wonderflow) [inline docs/en/developers/extensions/set-metrics.md:30]:
```suggestion
## Setting metrics policy
```

File: docs/en/developers/extensions/set-route.md [hunk 2]
```diff
@@ -6,6 +6,25 @@
...
+
+2. Update the chart repo
+    ```shell script
+    helm repo update
+    ```
+
+3. Install route trait controller
+    ```shell script
+    helm install --create-namespace -n vela-system routetrait oam.catalog/routetrait
+###Setting route policy
```
Comment (wonderflow) [inline docs/en/developers/extensions/set-route.md:27]:
```suggestion

### Setting route policy
```

File: docs/en/developers/extensions/set-autoscale.md [hunk 3]
```diff
@@ -3,6 +3,25 @@
 > Note: autoscale is one of the extension capabilities [installed from cap center](../cap-center.md),
 > please install it if you can't find it in `vela traits`.
 
+## Prerequisite
+Make sure auto-scaler trait controller installed in your cluster
```
Comment (zzxwill) [inline docs/en/developers/extensions/set-autoscale.md:7]:
```suggestion
Make sure auto-scaler trait controller is installed in your cluster
```

File: docs/en/developers/extensions/set-autoscale.md [hunk 4]
```diff
@@ -3,6 +3,25 @@
 > Note: autoscale is one of the extension capabilities [installed from cap center](../cap-center.md),
 > please install it if you can't find it in `vela traits`.
 
+## Prerequisite
```
Comment (zzxwill) [inline docs/en/developers/extensions/set-autoscale.md:6]: We can auto-generate this section by `hack/references/generate.go`, like other sections of each `set-xxx.md` docs.

Comment (wangyikewxgm) [reply docs/en/developers/extensions/set-autoscale.md:6]: Great idea, I think we can auto generate this section in the future : )

Comment (wonderflow) [reply docs/en/developers/extensions/set-autoscale.md:5]: yes, this is not high priority， we must finish it ASAP

File: go.mod [hunk 5]
```diff
@@ -77,6 +100,7 @@ require (
 	k8s.io/utils v0.0.0-20200603063816-c1c6865ac451
 	sigs.k8s.io/controller-runtime v0.6.2
 	sigs.k8s.io/controller-tools v0.2.4
+	sigs.k8s.io/structured-merge-diff/v3 v3.0.1-0.20200706213357-43c19bbb7fba // indirect
```
Comment (wonderflow) [inline go.mod:103]: why do we need to add these libs in this PR

**Review by wonderflow — APPROVED** [review]
GOOD job

**Pull request merged**